src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsReceiver.php

```diff
@@ -43,7 +44,7 @@ public function get(): iterable
         try {
             $sqsEnvelope = $this->connection->get();
         } catch (HttpException $e) {
-            throw new TransportException($e->getMessage(), 0, $e);
+            throw new RecoverableMessageHandlingException($e->getMessage(), 0, $e);
```

Comment:
I'd rather suggest to Create a new Exception that extends `TransportException` and Implements `RecoverableMessageHandlingExceptionInterface`.

Reply:
@jderusse I'm assuming it's the general `RecoverableExceptionInterface` I should implement? I'm thinking this new exception can live inside the Bridge-specific code as SQS is for now the only use case, e.g. in `namespace Symfony\Component\Messenger\Bridge\AmazonSqs\Exception`. Does that sound right or should it be more general?

Reply:
Indeed, I was thinking about the `RecoverableExceptionInterface`. 

But I'm not sure this will work. Look at the `Worker` class, nothing catch this exception.

The `RecoverableExceptionInterface` aims to be triggered by an handler and handled by a middleware. Throwing such exception in the Receiver will always stops the worker. 

I'd rather suggest to use the latest version of Async-Aws which uses the RetryiableHttpClient and automatically retries failed requests

Reply:
Thanks @jderusse, I was struggling to work out where it would be caught exactly, so this makes sense. We actually updated our codebase already so I'll try to get testing of that prioritised and the new version live as a higher priority.